test/benchmarks/scripts/run-benchmarks-ci.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+
+set -exuo pipefail
+
+NOW_ISO_8601=${NOW_ISO_8601:-$(date -u "+%Y-%m-%dT%H%M%SZ")}
+
+echo $(pwd)
+
+function setUp() {
+    echo "Setting CPU frequency to base frequency"
+
+    CPU_MODEL=$(lscpu | grep "Model name" | awk '{for(i=3;i<=NF;i++){printf "%s ", $i}; printf "\n"}')
+    if [ "${CPU_MODEL}" == "Intel(R) Xeon(R) CPU E3-1246 v3 @ 3.50GHz " ]
+    then
+        # could also use `nproc`
+        CORE_INDEX=7
+        BASE_FREQ="3.5GHz"
+    elif [ "${CPU_MODEL}" == "Intel(R) Core(TM) i7-6700 CPU @ 3.40GHz " ]
+    then
+        CORE_INDEX=7
+        BASE_FREQ="3.4GHz"
+    elif [ "${CPU_MODEL}" == "Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz " ]
+    then
+        CORE_INDEX=7
+        BASE_FREQ="3.6GHz"
+    else
+        >&2 echo "Cannot determine base frequency for CPU model [${CPU_MODEL}]. Please adjust the build script."
+        exit 1
+    fi
+    MIN_FREQ=$(cpufreq-info -l -c 0 | awk '{print $1}')
+    # This is the frequency including Turbo Boost. See also http://ark.intel.com/products/80916/Intel-Xeon-Processor-E3-1246-v3-8M-Cache-3_50-GHz
+    MAX_FREQ=$(cpufreq-info -l -c 0 | awk '{print $2}')
+
+    # set all CPUs to the base frequency
+    for (( cpu=0; cpu<=${CORE_INDEX}; cpu++ ))
+    do
+        sudo -n cpufreq-set -c ${cpu} --min ${BASE_FREQ} --max ${BASE_FREQ}
+    done
+
+    # Build cgroups to isolate microbenchmarks and JVM threads
+    echo "Creating groups for OS and microbenchmarks"
+    # Isolate the OS to the first core
+    sudo -n cset set --set=/os --cpu=0-1
+    sudo -n cset proc --move --fromset=/ --toset=/os
+
+    # Isolate the microbenchmarks to all cores except the first two (first physical core)
+    # On a 4 core CPU with hyper threading, this would be 6 cores (3 physical cores)
+    sudo -n cset set --set=/benchmark --cpu=2-${CORE_INDEX}
+}
+
+function benchmark() {
+    COMMIT_ISO_8601=$(git log -1 -s --format=%cI)
+    COMMIT_UNIX=$(git log -1 -s --format=%ct)
+
+    # TODO:
+    # [ -z "${NO_BUILD}" ] && ./mvnw clean package -DskipTests=true
+
+    RESULT_FILE=apm-agent-benchmark-results-${COMMIT_ISO_8601}.json
```

Comment:
This particular variable is also defined in the pipeline:
- https://github.com/elastic/apm-agent-nodejs/pull/1145/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R306

I guess we could add two different strategies:
- test/benchmarks/scripts/run-benchmarks-ci.sh uses arguments and the pipeline call the script `.ci/scripts/run-benchmarks.sh` with those parameters

```
  sh ".ci/scripts/run-benchmarks.sh ${RESULT_FILE}
```

Although I'm not sure if the npm might allow passing arguments somehow.

- If the env variable has been already defined then use the value:

```suggestion
    RESULT_FILE=${RESULT_FILE:-apm-agent-benchmark-results-${COMMIT_ISO_8601}.json}
```

What do you think?

Reply:
I'll just remove this definition, so it's only defined once.